sorc/ncep_post.fd/MDL2P.f

```diff
@@ -532,6 +533,68 @@ SUBROUTINE MDL2P(iostatusD3D)
+                 ! snow thermal profile. The model level above the 
+                 ! melting level will typically contain snow, but no 
+                 ! rain. Meanwhile, the model level below the melting
+                 ! level will contain rain. Importantly, model levels in
+                 ! this situation do not contain supercooled rain, but 
+                 ! interpolation onto the target pressure level will 
+                 ! yield supercooled rain. In this sense, the supercooled
+                 ! rain is an artifact of interpolation. Because
+                 ! supercooled water poses a hazard to aviation, we seek 
+                 ! to prevent supercooled water on pressure levels 
+                 ! when none actually exists on the adjacent model levels.
+                 ! 
+                 ! In the code below, we search for the condition in
+                 ! which this artifact occurs. Then, the previously 
+                 ! interpolated value of Qrain is simply replaced by the 
+                 ! value from the overlying (subfreezing) model level.
+                 ! The same approach is used for cloud water and cloud
+                 ! ice. However, for snow and graupel, either the value from
+                 ! the overlying model level or the interpolated value
+                 ! is used, whichever is greater. Since model-level weighting
+                 ! depends on the hydrometeor species, it is possible
+                 ! that the total condensate on the pressure level may be
+                 ! greater or less than that on both adjacent model levels
+                 ! (i.e., a slight artificial gain/loss of total condensate 
+                 ! is possible). If this behavior must be avoided, simply 
+                 ! use the values from the overlying model level for 
+                 ! all species.
+                 IF (MODELNAME == 'FV3R' .OR. MODELNAME == 'GFS') THEN
+                   IF ( TSL(I,J) <= TFRZ .AND.    & ! This pressure level is subfreezing and located just above a melting level;
+                        T(I,J,LL-1) <= TFRZ .AND. & !   i.e., the overlying model level is subfreezing,
+                        T(I,J,LL) > TFRZ ) THEN     !   but the underlying model level is above freezing.
+                          QW1(I,J) = MAX(QQW(I,J,LL-1),zero)    ! For cloud water, cloud ice, and rain,
+                          QI1(I,J) = MAX(QQI(I,J,LL-1),zero)    !   use the value from the overlying (subfreezing) model level. 
+                          QR1(I,J) = MAX(QQR(I,J,LL-1),zero)    ! 
+                          QS1(I,J) = MAX(QQS(I,J,LL-1),QS1(I,J))! For snow and graupel, use the value from the overlying
+                          QG1(I,J) = MAX(QQG(I,J,LL-1),QG1(I,J))!   level or the interpolated value, whichever is greater.
+                          C1D(I,J) = QG1(I,J)+QS1(I,J)+QR1(I,J)+QI1(I,J)+QW1(I,J) ! Recalculate total condensate
```

Comment:
@jaymes-kenyon I would suggest to add a change for preventing undefine grids involved in calculation, such as:
```
if (qg1(i,j)<spval .and. qs1(i,j,)<spval .and. qr1(i,j,)<spval .and. qi1(i,j,)<spval .and. qw1(i,j)<spval)  &
  c1d(i,j)= qg1(i,j)+qs1(i,j)+qr1(i,j)+qi1(i,j)+qw1(i,j)
```